src/modules/events/temperature_calibration.cpp

```diff
@@ -0,0 +1,369 @@
+	 */
+	Tempcal();
+
+	/**
+	 * Destructor, also kills task.
+	 */
+	~Tempcal();
+
+	/**
+	 * Start task.
+	 *
+	 * @return		OK on success.
+	 */
+	int		start();
+
+	static void do_temperature_calibration(int argc, char *argv[]);
+
+	void		task_main();
+
+	void print_status();
+
+	void exit() { _task_should_exit = true; }
+
+private:
+	bool	_task_should_exit = false;
+	int	_control_task = -1;		// task handle for task
+
+	/* Low pass filter for attitude rates */
+	//std::vector<math::LowPassFilter2p> _lp_roll_rate;
+	//std::vector<math::LowPassFilter2p> _lp_pitch_rate;
+	//std::vector<math::LowPassFilter2p> _lp_yaw_rate;
+};
+
+Tempcal::Tempcal():
+	SuperBlock(NULL, "Tempcal")
+	//_lp_roll_rate(SENSOR_COUNT_MAX, math::LowPassFilter2p(250.0f, 1.0f)),
+	//_lp_pitch_rate(SENSOR_COUNT_MAX, math::LowPassFilter2p(250.0f, 1.0f)),
+	//_lp_yaw_rate(SENSOR_COUNT_MAX, math::LowPassFilter2p(250.0f, 1.0f))
+{
+}
+
+Tempcal::~Tempcal()
+{
+
+}
+
+void Tempcal::task_main()
+{
+	// subscribe to relevant topics
+	int gyro_sub[SENSOR_COUNT_MAX];
+	float gyro_sample_filt[SENSOR_COUNT_MAX][4];
+	polyfitter<4> P[SENSOR_COUNT_MAX][3];
+	px4_pollfd_struct_t fds[SENSOR_COUNT_MAX] = {};
+	unsigned _hot_soak_sat[SENSOR_COUNT_MAX] = {};
+	unsigned num_gyro = orb_group_count(ORB_ID(sensor_gyro));
+	unsigned num_samples[SENSOR_COUNT_MAX] = {0};
+
+	bool _cold_soaked[SENSOR_COUNT_MAX] = {false};
+	bool _hot_soaked[SENSOR_COUNT_MAX] = {false};
+	bool _tempcal_complete[SENSOR_COUNT_MAX] = {false};
+	float _low_temp[SENSOR_COUNT_MAX];
+	float _high_temp[SENSOR_COUNT_MAX] = {0};
+	float _ref_temp[SENSOR_COUNT_MAX];
+
+	for (unsigned i = 0; i < num_gyro; i++) {
+		if (gyro_sub[i] < 0) {
+			gyro_sub[i] = orb_subscribe_multi(ORB_ID(sensor_gyro), i);
+		}
+	}
+
+	for (unsigned i = 0; i < num_gyro; i++) {
+		fds[i].fd = gyro_sub[i];
+		fds[i].events = POLLIN;
+	}
+
+	// initialize data structures outside of loop
+	// because they will else not always be
+	// properly populated
+	sensor_gyro_s gyro_data = {};
+	//uint16_t l = 0;
+
+	while (!_task_should_exit) {
+		int ret = px4_poll(fds, sizeof(fds) / sizeof(fds[0]), 1000);
+
+		if (ret < 0) {
+			// Poll error, sleep and try again
+			usleep(10000);
+			continue;
+
+		} else if (ret == 0) {
+			// Poll timeout or no new data, do nothing
+			continue;
+		}
+
+		for (unsigned i = 0; i < num_gyro; i++) {
+			if (_hot_soaked[i]) {
+				continue;
+			}
+
+			if (fds[i].revents & POLLIN) {
+				orb_copy(ORB_ID(sensor_gyro), gyro_sub[i], &gyro_data);
+
+				gyro_sample_filt[i][0] = gyro_data.x;//dat[l].GX;
+				gyro_sample_filt[i][1] = gyro_data.y;//dat[l].GY;
+				gyro_sample_filt[i][2] = gyro_data.z;//dat[l].GZ;
+				gyro_sample_filt[i][3] = gyro_data.temperature;
+
+				if (!_cold_soaked[i]) {
+					_cold_soaked[i] = true;
+					_low_temp[i] = gyro_sample_filt[i][3];	//Record the low temperature
+					_ref_temp[i] = gyro_sample_filt[i][3] + 12.0f;
+				}
+
+				num_samples[i]++;
+			}
+		}
+
+		//l++;
+		for (unsigned i = 0; i < 1; i++) {
```

Comment:
You fixed the one below but not this one :)
Also I have some changes, but cannot push to the branch. Can you give me access or should I open an new PR in the PX4 repo?